Fix deep highlighting clicks while node is selected

Cellv2 opened the pull request:
Fix #4 

This adds in a force rehighlighting function which triggers when clicking the doDeepUsage input element. It does this by keeping track of the currently selected box and utilising that when clicking the input

I would use boxFocused. however as the current listeners make this undefined when clicking off of the box via bgClick I thought it easier to just keep track of it this way

Cellv2 committed:
Fix deep highlighting clicks while node is selected

Cellv2 committed:
Remove vscode settings

Cellv2 commented:
Sorry, just noticed there's no gitignore so .vscode\settings.json crept in

I've updated the PR to remove this

Cellv2 committed:
Update code formatting

Update the code formatting to be in line with the codebase

kukuminer commented:
Hey @Cellv2, thanks for your PR!
I had a look and I noticed that we can actually use `boxFocused` instead of a new `currentBox` variable. We can change line 280 to 
`boxFocused = box` from `.. = box.id`
to accommodate.
Resetting `boxFocused` in the `boxClick` function (line 273) also prevents a specific bug (clicking another box while one is in focus and after clicking the checkbox)

I'll put some comments on your PR to indicate the changes that I think would simplify this PR as I outlined above.

I hope what I said is clear, if you have any questions please don't hesitate to ask!

kukuminer commented on script.js:
```diff
@@ -1,6 +1,8 @@
 var boxFocused = undefined;
+let currentBox = null;
```
We can use `boxFocused` for this purpose, so remove this variable and refactor all uses

kukuminer commented on script.js:
```diff
@@ -145,6 +154,7 @@ function boxHover(event)
 	if(!boxFocused)
 	{
 		highlight(event.target);
+		currentBox = event.target;
```
This isn't necessary since they can't click the checkbox while hovering on a mod box, but we can leave it in for future development (just make sure to refactor)

Cellv2 replied:
Really isn't needed as you said, but left it in anyway

kukuminer replied:
I was actually wrong about this and didn't realize the implications- this prevents the box from un-highlighting after the cursor leaves it. We'll need to remove this line for it to work as intended, sorry about that

Cellv2 commented:
@kukuminer Sure thing, honestly I did a quick hack to get it put together, I probably should have read the code more closely 😄 

Will take a look through and update the PR

kukuminer reviewed (changes requested):
Don't forget to change line 281: 
`boxFocused = box.id` => `boxFocused = box`

Thanks again for your PR!

Cellv2 committed:
Refactor box variables for #5

Cellv2 commented:
@kukuminer PR updated with requested changes

Also, the bug you mentioned on like 273 is exactly what I hit when trying to originally refactor, but I just didn't look closely enough at what is being passed around to notice the quick change of box.id to box on line 279, thanks for pointing it out

Cellv2 committed:
Formatting

Remove additional blank line

kukuminer commented on script.js:
```diff
@@ -145,6 +153,7 @@ function boxHover(event)
 	if(!boxFocused)
 	{
 		highlight(event.target);
+		boxFocused = event.target;
```
I didn't notice but this actually stops the box from un-highlighting. I think we just remove this line for now and we can add it back in a different way if we need to use this feature later. Sorry for the confusion

Cellv2 committed:
Fix box not deselecting when mousing away

Cellv2 commented:
@kukuminer Sorry I shouldn't code when I am so tired. This was something I noticed originally and was another reason I added the second variable (didn't quite look through enough to figure out why). I've removed it

kukuminer reviewed (approved):
Looks good! Did some quick testing and it works. Thanks so much for the PR and for the changes 😄
